### PR TITLE
Remove unnecessary `clone()`s in bootstrap

### PR DESCRIPTION
**Pull request by jyn514**
The performance difference is negligible, but it makes me feel better.

r? @Mark-Simulacrum

### CODE REVIEW

File: src/bootstrap/config.rs [hunk 1]
```diff
@@ -671,6 +650,28 @@ impl Config {
...
+            config.llvm_version_suffix = llvm.version_suffix;
+            config.llvm_clang_cl = llvm.clang_cl;
+
+            config.llvm_cflags = llvm.cflags;
+            config.llvm_cxxflags = llvm.cxxflags;
+            config.llvm_ldflags = llvm.ldflags;
+            set(&mut config.llvm_use_libcxx, llvm.use_libcxx);
+            config.llvm_use_linker = llvm.use_linker;
+            config.llvm_allow_old_toolchain = llvm.allow_old_toolchain;
+            config.llvm_from_ci = llvm.download_ci_llvm.unwrap_or(false);
```
Comment (Mark-Simulacrum) [inline src/bootstrap/config.rs:673]: Why was this moved after the checking? I preferred the old order, and in any case, this won't work because llvm_from_ci will always be false in the if above.

Comment (jyn514) [reply src/bootstrap/config.rs:673]: Hmm, I moved it because otherwise there was a borrow error that `llvm` had been moved. I could either move `             config.llvm_from_ci = llvm.download_ci_llvm.unwrap_or(false);` above or change the checking to use `config.llvm` instead.

Comment (jyn514) [reply src/bootstrap/config.rs:644]: Ok, I changed it to use `config.llvm_*` instead.

File: src/bootstrap/config.rs [hunk 2]
```diff
@@ -654,19 +655,21 @@ impl Config {
...
-                check_ci_llvm!(llvm.clang_cl);
-                check_ci_llvm!(llvm.version_suffix);
-                check_ci_llvm!(llvm.cflags);
-                check_ci_llvm!(llvm.cxxflags);
-                check_ci_llvm!(llvm.ldflags);
                 check_ci_llvm!(llvm.use_libcxx);
-                check_ci_llvm!(llvm.use_linker);
                 check_ci_llvm!(llvm.allow_old_toolchain);
 
+                // N.B. these use `config` because they were moved above
```
Comment (Mark-Simulacrum) [inline src/bootstrap/config.rs:663]: I don't think we should complicate the code for the cost of a few clones -- remember that bootstrap isn't even compiled in release mode, performance *does not matter*.

Comment (jyn514) [reply src/bootstrap/config.rs]: Ok, I reverted the changes.

**Comment by Mark-Simulacrum** [comment]
r=me with everything squashed

**Comment by jyn514** [comment]
@bors r=Mark-Simulacrum rollup

Merge conflict was a trivial one-line change.
